module-system/module-implementations/sov-accounts/src/lib.rs

```diff
@@ -75,3 +84,62 @@ impl<C: sov_modules_api::Context> sov_modules_api::Module for Accounts<C> {
         }
     }
 }
+
+#[cfg(feature = "arbitrary")]
+impl<'a, C> arbitrary::Arbitrary<'a> for Account<C>
+where
+    C: Context,
+    C::Address: arbitrary::Arbitrary<'a>,
+{
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        let addr = u.arbitrary()?;
+        let nonce = u.arbitrary()?;
+        Ok(Self { addr, nonce })
+    }
+}
+
+#[cfg(feature = "arbitrary")]
+impl<'a, C> arbitrary::Arbitrary<'a> for AccountConfig<C>
+where
+    C: Context,
+    C::PublicKey: arbitrary::Arbitrary<'a>,
+{
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        // TODO we might want a dedicated struct that will generate the private key counterpart so
+        // payloads can be signed and verified
+        Ok(Self {
+            pub_keys: u.arbitrary_iter()?.collect::<Result<_, _>>()?,
+        })
+    }
+}
+
+#[cfg(feature = "arbitrary")]
+impl<'a, C> Accounts<C>
+where
+    C: Context,
+    C::Address: arbitrary::Arbitrary<'a>,
+    C::PublicKey: arbitrary::Arbitrary<'a>,
+{
+    /// Creates an arbitrary set of accounts and stores it under `working_set`.
+    pub fn arbitrary_workset(
+        u: &mut arbitrary::Unstructured<'a>,
+        working_set: &mut WorkingSet<C::Storage>,
+    ) -> arbitrary::Result<Self> {
+        use sov_modules_api::Module;
+
+        let address = u.arbitrary()?;
+        let config: AccountConfig<C> = u.arbitrary()?;
+        let prefix: sov_state::Prefix = u.arbitrary()?;
+        let accounts = Self {
+            address,
+            public_keys: sov_state::StateMap::new(prefix.clone()),
+            accounts: sov_state::StateMap::new(prefix),
+        };
+
+        accounts
+            .genesis(&config, working_set)
+            .map_err(|_| arbitrary::Error::IncorrectFormat)?;
+
+        Ok(accounts)
+    }
+}
```

Comment:
`Arbitrary` for modules is super cool to see, it would simplify certain testing patterns –not just fuzzing– quite a lot.  🎉

We have to be especially careful around prefix conflicts for `State[Map|Value|Vec]`. In this case, the two maps `public_keys` and `accounts` use the same prefix which can cause key conflicts. I'd propose we always use `Default::default` when initializing modules inside `arbitrary_workset`, and only then we call `genesis` with arbitrary configuration.

Reply:
That makes sense! I did like that due to lack of knowledge of the code base; I was initially under the misconception that we need to share these prefixes across different maps, but now I know that they *must* be different.